### PR TITLE
fix: set the Accessory activation policy so the menu bar app stays out of the Dock

### PR DESCRIPTION
**Pull request by fantasywind**
## Problem

Running `cswap menubar` under a framework Python (e.g. the python.org installer, where the interpreter execs `Python.app/Contents/MacOS/Python`) parks a permanent **"Python" icon in the Dock** for as long as the menu bar runs. Quitting it from the Dock doesn't help either — anyone running the menu bar from a KeepAlive launchd agent just gets it respawned.

## Root cause

rumps (0.4.0) never sets an `NSApplication` activation policy, so macOS treats the process as a regular foreground app: Dock icon, Cmd-Tab entry, the works. A menu-bar-only app should be an *accessory* app.

## Fix

Set `NSApplicationActivationPolicyAccessory` at the top of `menubar.run()`, before the rumps app starts. Accessory (rather than Prohibited) is deliberate: the "Add from setup-token" flow activates the app to show `rumps.Window` dialogs, and accessory apps can still be activated and present windows — they just never appear in the Dock or the Cmd-Tab switcher.

The import stays inside `run()` alongside the lazy `rumps` import, so the module remains import-safe without AppKit and the pure-helper tests keep running on Linux/Windows CI.

## Verification

On macOS (Darwin 25.5.0, Python 3.12.8 framework build), with the menu bar relaunched from its launchd agent:

- `lsappinfo info` for the process now reports `type="UIElement"` (was a regular app before)
- System Events reports `background only = true`
- No Dock icon; the ⇄ status item, dropdown menu, account switching, and the add-from-token dialogs all still work
- `uv run pytest`: no new failures vs `main` (identical pass/fail set, 1346 passed locally)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/claude_swap/menubar.py [hunk 1]
```diff
@@ -369,6 +369,16 @@ def run(switcher) -> int:
     """Entry point for ``cswap --menubar``. Blocks until the user quits."""
     import rumps  # lazy: optional dependency, imported only when launching
 
+    import AppKit
+
```
[the remark this answers is not in the document]
Comment (fantasywind) [reply src/claude_swap/menubar.py]: Good catch — confirmed and fixed in f5f8dc1. The friendly install hint in `cli.py` was indeed dead code: the guarded `from claude_swap.menubar import run` is import-safe by design, so a missing extra crashed later inside `run()` with a raw `ModuleNotFoundError`. `run()` now catches the lazy import failure and re-raises it as `ClaudeSwitchError` with the same install hint, which the CLI already renders cleanly (verified: `cswap menubar` without rumps now prints `Error: Menu bar mode requires 'rumps'. Install with: pip install 'claude-swap[menubar]'` and exits 1). Also added a cross-platform test that blocks the rumps import via a `None` `sys.modules` entry — first coverage of this error path, and it runs on Linux/Windows CI too since nothing AppKit is touched before the raise.

**Comment by realiti4** [comment]
Thanks for the fix, looks good. I have merged it.
